addons/controllers/cpi/cpiconfig_utils.go

```diff
@@ -0,0 +1,307 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capvv1beta1 "sigs.k8s.io/cluster-api-provider-vsphere/api/v1beta1"
+	clusterapiv1beta1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/constants"
+	pkgtypes "github.com/vmware-tanzu/tanzu-framework/addons/pkg/types"
+	"github.com/vmware-tanzu/tanzu-framework/addons/pkg/util"
+	cpiv1alpha1 "github.com/vmware-tanzu/tanzu-framework/apis/cpi/v1alpha1"
+)
+
+// mapCPIConfigToDataValuesNonParavirtual generates CPI data values for non-paravirtual modes
+func (r *CPIConfigReconciler) mapCPIConfigToDataValuesNonParavirtual( // nolint
+	ctx context.Context,
+	cpiConfig *cpiv1alpha1.CPIConfig, cluster *clusterapiv1beta1.Cluster) (*CPIDataValues, error,
+) { // nolint:whitespace
```

Comment:
Better not to use `nolint` unless we have no other choice

Reply:
it will complain about that the cyclomatic complexity for this function is too large.